test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,204 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
 		expectedError: true,
 	},
 }
+
+var OVHCloudSDTestCases = []scrapeCRDTestCase{
+	{
+		name: "Valid OVHCloudSDConfig",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceDedicatedServer,
+					Endpoint:          ptr.To("https://api.ovh.com/endpoint"),
+					RefreshInterval:   ptr.To(monitoringv1.Duration("30s")),
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Valid OVHCloudSDConfig without optional fields",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceDedicatedServer,
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Valid ApplicationKey",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceVPS,
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Empty ApplicationKey",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceVPS,
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Missing ApplicationKey",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceVPS,
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid Service field",
```

Comment:
(nit) Just to make it explicit, add another test case for `OVHServiceDedicatedServer`. (I can see that you have already added it as a part of another test case.)